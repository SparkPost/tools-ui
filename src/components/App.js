import React, { Component } from 'react';
import { connect } from 'react-redux';
import classnames from 'classnames';
import Nav from 'components/nav/Nav';
import Footer from 'components/footer/Footer';
import { Meta, DefaultMeta } from 'components/Meta';
import { checkLogin } from 'actions/auth';
import { trackPageView } from 'actions/mixpanel';

export class App extends Component {

  // need to use this lifecycle stage to make sure that
  // the cookie is read/token in place in time
  componentWillMount() {
    this.props.checkLogin();
  }

  render() {
    const { children, location, loggedIn } = this.props;
    return (
      <div className={classnames('pageWrapper', { 'pageWrapper--loggedIn': loggedIn })}>
        <DefaultMeta />
        <Meta location={location} />
        <div className='container container--tool'>
          {children}
        </div>
        <Footer />
        <Nav location={location} />
      </div>
    );
  }
}

const mapStateToProps = ({ auth }) => ({ loggedIn: auth.loggedIn });
export default connect(mapStateToProps, { checkLogin, trackPageView })(App);
