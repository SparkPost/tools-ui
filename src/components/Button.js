import React from 'react';
import { Link } from 'react-router';
import classNames from 'classnames';
const noop = () => {};

const mapPropsToClasses = ({ type, size, accent, fullWidth, icon, states, extraClasses }) => (
  classNames('button', {
    [`button--${type}`]: type,
    [`button--${size}`]: size,
    [`button--${accent}Accent`]: accent,
    'button--full': fullWidth,
    'button--icon': icon
  }, states, extraClasses)
);

const ActionButton = (props) => {
  const { action = noop, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <button onClick={action} className={classes}>
      {children}
    </button>
  );
};

const LinkButton = (props) => {
  const { to = null, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <Link to={to} className={classes}>
      {children}
    </Link>
  );
};

export { ActionButton, LinkButton };
