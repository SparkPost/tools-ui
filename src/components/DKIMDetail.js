import React, { Component } from 'react';
import { Link } from 'react-router';
import axios from 'axios';
import config from '../config';

import HeaderTable from './DKIMDetailHeaderTable';
import SignatureTable from './DKIMSignatureTable';

export default class DKIMDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: props.email,
      error: null,
      detail: null,
      loading: false // should this just be true at first?
    }
  }

  componentDidMount() {
    this.getDetail();
  }

  getDetail() {
    const { email, detailId } = this.props;
    this.setState({ loading: true });
    return axios.get(`${config.apiBase}/messaging-tools/validations/${email}/${detailId}`)
      .then(({ data }) => {
        const { results } = data;
        this.setState({
          detail: results,
          loading: false
        });
      }, ({ response = null }) => {
        this.setState({
          error: response ? response.data.errors[0] : true,
          loading: false
        });
      });
  }

  render() {
    const { detail = {}, error } = this.state;
    return (
      <div>
        <div className='button-header'>
          <Link to={`/dkim/results/${this.props.email}`} className='button button--muted'><i className='fa fa-arrow-left'></i> Back</Link>
          <button className='button button--muted'><i className='fa fa-share-alt'></i> Share</button>
        </div>
        {error && <p>An error occurred getting details</p>}
        <HeaderTable detail={detail} />
        <SignatureTable sigs={detail && detail.sigs} />
      </div>
    )
  }
}
