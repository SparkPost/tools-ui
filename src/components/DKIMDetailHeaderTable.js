import React from 'react';

export default ({ detail }) => {
  if (!detail) {return null;}
  return (
    <table>
      <tbody>
        <tr>
          <th>Subject</th>
          <td>{detail.subject}</td>
        </tr>
        <tr>
          <th>Sender</th>
          <td>{detail.header_from}</td>
        </tr>
        <tr>
          <th>Date</th>
          <td>{new Date(detail.received).toLocaleString()}</td>
        </tr>
        <tr>
          <th>DKIM Status</th>
          <td>{detail.result ? 'Passed' : 'Failed'}</td>
        </tr>
      </tbody>
    </table>
  );
};
