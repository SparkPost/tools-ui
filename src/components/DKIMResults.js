import React, { Component } from 'react';
import { Link } from 'react-router';
import axios from 'axios';
import config from '../config';

export default class extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      results: [],
      loading: false // should this just be true at first?
    }
  }

  componentDidMount() {
    this.getResults();
  }

  getResults() {
    const { email } = this.props;
    this.setState({ loading: true });
    return axios.get(`${config.apiBase}/messaging-tools/validations/${email}`)
      .then(({ data }) => {
        const { results } = data;
        this.setState({
          results,
          loading: false
        });
      }, ({ response }) => {
        this.setState({
          error: response.data.errors[0],
          loading: false
        });
      });
  }

  renderError() {
    const { error } = this.state;
    if (!error) return null;
    return (
      <div className='error'>
        <i className='fa fa-exclamation-circle'></i>
        {error.message}
      </div>
    )
  }

  renderHeader() {
    const { email } = this.props;
    return (
      <div>
        <p>
          <strong>Results for Test Address:</strong>
          <br/>{email}
        </p>
        <button onClick={() => this.getResults()}>Refresh</button>
        <button>Share</button>
      </div>
    );
  }

  renderTable() {
    const { loading, results } = this.state;
    if (loading) return null;
    return (
      <table>
        <thead>
          <tr>
            <th>Subject</th>
            <th>DKIM Status</th>
            <th>Sender</th>
            <th>Time Delivered</th>
          </tr>
        </thead>
        <tbody>
        {
          (results.length === 0)
          ? this.renderNoMessages()
          : results.map(this.renderRow.bind(this))
        }
        </tbody>
      </table>
    );
  }

  renderNoMessages() {
    return (
      <tr className='no-messages'><td colSpan='4'>No messages have been recieved to this test address.</td></tr>
    );
  }

  renderRow({ id, subject, result, header_from, received }) {
    const { email } = this.props;
    return (
      <tr key={id}>
        <td><Link to={`/dkim/results/${email}/${id}`}>{subject}</Link></td>
        <td>{result ? 'Passed' : 'Failed'}</td>
        <td>{header_from}</td>
        <td>{new Date(received).toLocaleString()}</td>
      </tr>
    );
  }

  render() {
    const { error } = this.state;
    return (
      <div>
        {this.renderHeader()}
        {error ? this.renderError() : this.renderTable()}
      </div>
    )
  }
}
