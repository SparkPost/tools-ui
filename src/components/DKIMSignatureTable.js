import React from 'react';

const Row = ({ sig }) => {
  const { s, d, t, result } = sig;
  return (
    <tr>
      <td>{result ? 'Passed' : 'Failed'}</td>
      <td>{s}</td>
      <td>{d}</td>
      <td>{t || 'N/A'}</td>
    </tr>
  );
};

export default ({ sigs = null }) => {
  if (!sigs) {return null;}

  return (
    <table>
      <thead>
        <tr>
          <th>Status</th>
          <th>DKIM Selector (s=)</th>
          <th>Signing Domain (d=)</th>
          <th>Timestamp (t=)</th>
        </tr>
      </thead>
      <tbody>
        {sigs.map((sig) => <Row sig={sig} key={sig.s} />)}
      </tbody>
    </table>
  );
};
