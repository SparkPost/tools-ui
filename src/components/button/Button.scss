@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

@import "~@sparkpost/styles/src/assets/scss/components/button";

.button {
  padding: 0 rem(12);
  white-space: nowrap;
}

.actionLink {
  display: inline-block;
  padding: 0 rem(6);

  color: color(orange);
  font-weight: 400;
  font-size: rem(14);

  &:hover, &:active, &:focus {
    cursor: pointer;
    text-decoration: none;
    color: color(orange--dark);
    background: grayscale(9);
  }

  @media screen and (min-width: break(md)) {
    padding: 0 rem(12);
    font-size: rem(15);
  }
}

.backLink {
  display: inline-block;
  margin-bottom: rem(9);
  font-size: rem(15);

  .fa {
    transition: 0.15s;
  }

  &:hover {
    text-decoration: none;

    .fa {
      transform: translate(-1px, 0);
    }
  }
}
