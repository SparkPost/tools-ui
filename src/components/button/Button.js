import React from 'react';
import { Link } from 'react-router';
import Icon from 'components/Icon';
import { HoverPopover } from 'components/popover/Popover';
import config from 'config/index';
import getCurrentUrl from 'helpers/getCurrentUrl';
import classNames from 'classnames';
const noop = () => {};

import './Button.scss';

const mapPropsToClasses = ({ type, size, accent, fullWidth, icon, states, extraClasses }) => (
  classNames('button', {
    [`button--${type}`]: type,
    [`button--${size}`]: size,
    [`button--${accent}Accent`]: accent,
    'button--full': fullWidth,
    'button--icon': icon
  }, states, extraClasses)
);

const ActionButton = (props) => {
  const { action = noop, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <button onClick={action} className={classes}>
      {children}
    </button>
  );
};

const LinkButton = (props) => {
  const { to = null, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <Link to={to} className={classes}>
      {children}
    </Link>
  );
};

/**
 * Produces an orange link
 * Can be used to link or execute an action
 */
const ActionLink = (props) => {
  const { to = null, external = null, onClick = null, title = '', children } = props;

  if (external) {
    return <a href={external} className='actionLink' title={title}>{children}</a>;
  }

  return <Link to={to} onClick={onClick} className='actionLink' title={title}>{children}</Link>;
};

/**
 * Produces a blue link to indicate browser back
 */
const BackLink = ({ to = null, title = '' }) => <Link to={to} className='backLink' title={title}><Icon name='chevron-left' /> Back</Link>;

/**
 * Produces a Save Results action link
 */
const SaveResultsLink = () => {
  const currentUrl = getCurrentUrl(location);
  return (
    <HoverPopover placement='top' size='m' text='Create a free SparkPost account or login into your account to save results'>
      <ActionLink external={`${config.appUrl}/sign-up?return=${currentUrl}`} className='actionLink' title='Save Results'>Save Results</ActionLink>
    </HoverPopover>
  );
};

const SpLoginLink = ({ location = {}, classes, children }) => {
  const currentUrl = getCurrentUrl(location);
  const linkClasses = classNames('sp-sign-in', classes);
  return (
    <a href={`${config.appUrl}/auth?return=${currentUrl}`} title='Login' className={linkClasses}>{children}</a>
  );
};

const SpSignUpLink = ({ location = {}, classes, children }) => {
  const currentUrl = getCurrentUrl(location);
  const linkClasses = classNames('sp-sign-up', classes);
  return (
    <a href={`${config.appUrl}/sign-up?return=${currentUrl}&src=SP-Tools`} title='Sign Up' className={linkClasses}>{children}</a>
  );
};

export { ActionButton, LinkButton, ActionLink, BackLink, SpLoginLink, SpSignUpLink, SaveResultsLink };
