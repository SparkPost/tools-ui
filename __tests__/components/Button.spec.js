import React from 'react';
import renderer from 'react-test-renderer';

import { ActionButton, LinkButton } from '../../src/components/Button';

describe('snapshot tests', () => {
  test('ActionButton will render correctly with all options', () => {
    const someFunction = () => {};
    expect(renderer.create(<ActionButton
      action={someFunction}
      type='muted'
      size='l'
      fullWidth
      icon
      accent='magenta'
      states={['is-disabled', 'is-awesome']}
      extraClasses={['has-error', 'best-button-ever']}
    >This is an action button</ActionButton>)).toMatchSnapshot();
  });

  test('ActionButton will render correctly with no options', () => {
    const someFunction = () => {};
    expect(renderer.create(<ActionButton>ACTION</ActionButton>)).toMatchSnapshot();
  });

  test('LinkButton will render correctly with all options', () => {
    expect(renderer.create(<LinkButton
      to='some url'
      type='muted'
      size='l'
      fullWidth
      icon
      accent='magenta'
      states={['is-disabled', 'is-awesome']}
      extraClasses={['has-error', 'best-button-ever']}
    >This is a link button</LinkButton>)).toMatchSnapshot();
  });

  test('LinkButton will render correctly with no options', () => {
    expect(renderer.create(<LinkButton>LINK</LinkButton>)).toMatchSnapshot();
  });
});
