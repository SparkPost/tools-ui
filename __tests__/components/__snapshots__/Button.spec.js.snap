exports[`snapshot tests ActionLink will render correctly with all options 1`] = `
<a
  className="actionLink"
  onClick={[Function]}
  style={Object {}}
  title="a title">
  Action link text
</a>
`;

exports[`snapshot tests ActionLink will render correctly with no options 1`] = `
<a
  className="actionLink"
  onClick={[Function]}
  style={Object {}}
  title="" />
`;

exports[`snapshot tests BackLink will render correctly with all options 1`] = `
<a
  className="backLink"
  onClick={[Function]}
  style={Object {}}
  title="a title">
  <i
    className="fa fa-chevron-left " />
   Back
</a>
`;

exports[`snapshot tests BackLink will render correctly with no options 1`] = `
<a
  className="backLink"
  onClick={[Function]}
  style={Object {}}
  title="">
  <i
    className="fa fa-chevron-left " />
   Back
</a>
`;

exports[`snapshot tests Button will render a Link with a url 1`] = `
<a
  className="button"
  onClick={[Function]}
  style={Object {}}>
  ACTION
</a>
`;

exports[`snapshot tests Button will render correctly with all options 1`] = `
<button
  className="button button--muted button--l button--magentaAccent button--full button--icon is-disabled is-awesome has-error best-button-ever"
  onClick={[Function]}>
  This is an action button
</button>
`;

exports[`snapshot tests Button will render correctly with no options 1`] = `
<button
  className="button"
  onClick={[Function]}>
  ACTION
</button>
`;

exports[`snapshot tests SpLoginLink will render correctly with all options 1`] = `
<a
  className="sp-sign-in a-list of-classes"
  href="https://TEST.sparkpost.com/auth?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
  onClick={[Function]}
  title="Login">
  Log In
</a>
`;

exports[`snapshot tests SpLoginLink will render correctly with no options 1`] = `
<a
  className="sp-sign-in"
  href="https://TEST.sparkpost.com/auth?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
  onClick={[Function]}
  title="Login" />
`;

exports[`snapshot tests SpSignUpLink will render correctly with all options 1`] = `
<a
  className="sp-sign-up a-list of-classes"
  href="https://TEST.sparkpost.com/join?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
  onClick={[Function]}
  title="Sign Up">
  Sign Up
</a>
`;

exports[`snapshot tests SpSignUpLink will render correctly with no options 1`] = `
<a
  className="sp-sign-up"
  href="https://TEST.sparkpost.com/join?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
  onClick={[Function]}
  title="Sign Up" />
`;
